tests/test_kb_auth.py

```diff
@@ -0,0 +1,123 @@
+import asyncio
+import pytest
+from staging_service.kb_auth_client import InvalidUserError, KBaseAuth, InvalidTokenError, _get
+from tests.test_utils import bootstrap_config
+import os
+from unittest.mock import patch
+
+config = bootstrap_config()
+AUTH_URL = config["staging_service"]["AUTH_URL"]
+
+# NOTE: These tests are intended to be run against the auth service at
+# https://ci.kbase.us/services/auth
+# Any unexpected failures should check the auth service, the given test
+# token, and whether the test user exists or not.
+
+TEST_TOKEN = os.environ.get("KB_TEST_TOKEN")
+TEST_USER = os.environ.get("KB_TEST_USER")
+VALID_TEST_USER = "narrativetest"
+NOT_REAL_USER = "please_do_not_ever_make_this_username_what_is_wrong_with_you"
+INVALID_USER = "_(__)_"
+
+SKIP_AUTH_TESTS = not TEST_TOKEN or not TEST_USER
+
+
+@pytest.fixture
+async def auth_client():
+    """
+    A fixture that makes a default auth client with a shiny new cache.
+    Avoids the extra lag from using KBaseAuth.create.
+    """
+    return await KBaseAuth.create(AUTH_URL)
```

Comment:
I think this still creates a client per test otherwise

```suggestion
@pytest.fixture(scope = "module")
async def auth_client():
    """
    A fixture that makes a default auth client with a shiny new cache.
    Avoids the extra lag from using KBaseAuth.create.
    """
    return await KBaseAuth.create(AUTH_URL)
```

Reply:
Yeah, that's the idea. It should make a new client with a new cache. Honestly, I started with calling `__init__` directly, but then realized I'd like to keep that in the black box it's in. This remains a convenience.

Reply:
Ok, the 

> Avoids the extra lag from using KBaseAuth.create.

made me think you were trying for performance here hence the module scope suggestion, but got it.

I guess just that lines needs deletion now

Reply:
Deleted.

Reply:
:+1: